### PR TITLE
The Witness: Implement "Variety" puzzles mode

### PR DESCRIPTION
**Pull request by NewSoupVi**
Implements Umbra Moruka's [Variety Randomizer](https://github.com/Penguin12335/Variety-Randomiser) into Archipelago as an alternate puzzle generation mode to Sigma Normal and Sigma Expert.

This mostly just involves adding a new `WitnessLogicVariety.txt` file and some minor tweaks here and there.

Thank you @Exempt-Medic for making the "WitnessLogicVariety.txt" file.

A [compatible client branch](https://github.com/NewSoupVi/The-Witness-Randomizer-for-Archipelago/tree/variety_randomizer) exists, I have not uploaded an .exe yet but can do so any time (it seems to be functional so far, I have played a full seed of it)

### CODE REVIEW

**Comment by NewSoupVi** [comment]
This is ready to be reviewed, I just don't want it to be merged yet because I'm still testing it.

File: worlds/witness/options.py [hunk 1]
```diff
@@ -177,11 +177,16 @@ class VictoryCondition(Choice):
...
+    "Sigma Normal" randomizes puzzles close to their original mechanics and difficulty.
+    "Sigma Expert" is an entirely new experience with extremely difficult random puzzles. Do not underestimate this mode, it is brutal.
+    "None" means that the puzzles are unchanged from the original game.
+    "Umbra Variety" focuses on unique symbol combinations not featured in the original game. It is harder than Sigma Normal, but easier than Sigma Expert.
     """
     display_name = "Puzzle Randomization"
     option_sigma_normal = 0
     option_sigma_expert = 1
     option_none = 2
+    option_umbra_variety = 3
```
[the remark this answers is not in the document]
Comment (NewSoupVi) [reply worlds/witness/options.py:189]: I'll try it but I think it's numbers :(

[the remark this answers is not in the document]
Comment (NewSoupVi) [reply worlds/witness/options.py:189]: nvm. Epic

**Comment by NewSoupVi** [comment]
A lot of seeds have been played by players now :) So I'm ready for this to be reviewed

File: worlds/witness/data/static_items.py [hunk 2]
```diff
@@ -13,6 +13,22 @@
 # item list during get_progression_items.
 _special_usefuls: List[str] = ["Puzzle Skip"]
 
+ALWAYS_GOOD_SYMBOL_ITEMS : Set[str] = {"Dots", "Black/White Squares", "Symmetry", "Shapers", "Stars"}
```
Comment (Exempt-Medic) [inline worlds/witness/data/static_items.py:16]:
```suggestion
ALWAYS_GOOD_SYMBOL_ITEMS: Set[str] = {"Dots", "Black/White Squares", "Symmetry", "Shapers", "Stars"}
```

File: worlds/witness/data/static_logic.py [hunk 3]
```diff
@@ -260,28 +262,35 @@ def get_parent_progressive_item(item_name: str) -> str:
...
+
+
 def __getattr__(name):
     if name == "vanilla":
         return get_vanilla()
     elif name == "sigma_normal":
         return get_sigma_normal()
     elif name == "sigma_expert":
         return get_sigma_expert()
+    elif name == "umbra_variety":
```
Comment (Exempt-Medic) [inline worlds/witness/data/static_logic.py:292]:
```suggestion
    if name == "umbra_variety":
```

**Review by Exempt-Medic — APPROVED** [review]
Changes all LGTM. Tested hundreds of generations, played through five games, checked the early good item code specifically through print debugging. Two very minor suggestions

**Comment by NewSoupVi** [comment]
Drafting this because I want to add a unit test for this.

File: worlds/witness/data/WitnessLogicVariety.txt [hunk 4]
```diff
@@ -0,0 +1,1222 @@
...
+158421 - 0x33AF5 (Back Row 1) - True - Symmetry & Triangles
+158422 - 0x33AF7 (Back Row 2) - 0x33AF5 - Triangles
+158423 - 0x09F6E (Back Row 3) - 0x33AF7 - Symmetry & Triangles
+158424 - 0x09EAD (Trash Pillar 1) - True - Triangles & Arrows
+158425 - 0x09EAF (Trash Pillar 2) - 0x09EAD - Triangles & Arrows
+
+Mountain Floor 1 At Door (Mountain Floor 1) - Mountain Floor 2 - 0x09E54:
+Door - 0x09E54 (Exit) - 0x09EAF & 0x09F6E & 0x09E6B & 0x09E7B
+
+Mountain Floor 2 (Mountain Floor 2) - Mountain Floor 2 Light Bridge Room Near - 0x09FFB - Mountain Floor 2 Beyond Bridge - 0x09E86 - Mountain Floor 2 At Door - 0x09ED8 & 0x09E86 - Mountain Pink Bridge EP - TrueOneWay:
```
Comment (Exempt-Medic) [inline worlds/witness/data/WitnessLogicVariety.txt:1031]:
```suggestion
Mountain Floor 2 (Mountain Floor 2) - Mountain Floor 2 Light Bridge Room Near - 0x09FFB - Mountain Floor 2 Beyond Bridge - 0x09E86 - Mountain Floor 2 Above The Abyss - 0x09ED8 & 0x09E86 - Mountain Pink Bridge EP - TrueOneWay:
```

File: worlds/witness/data/WitnessLogicVariety.txt [hunk 5]
```diff
@@ -0,0 +1,1222 @@
...
+158426 - 0x09FD3 (Near Row 1) - True - Stars & Stars + Same Colored Symbol & Colored Squares
+158427 - 0x09FD4 (Near Row 2) - 0x09FD3 - Stars & Stars + Same Colored Symbol & Triangles
+158428 - 0x09FD6 (Near Row 3) - 0x09FD4 - Stars & Stars + Same Colored Symbol & Colored Squares & Eraser
+158429 - 0x09FD7 (Near Row 4) - 0x09FD6 - Stars & Stars + Same Colored Symbol & Shapers & Eraser
+158430 - 0x09FD8 (Near Row 5) - 0x09FD7 - Symmetry & Triangles
+Door - 0x09FFB (Staircase Near) - 0x09FD8
+
+Mountain Floor 2 At Door (Mountain Floor 2) - Mountain Floor 2 Elevator Room - 0x09EDD:
```
Comment (Exempt-Medic) [inline worlds/witness/data/WitnessLogicVariety.txt:1039]:
```suggestion
Mountain Floor 2 Above The Abyss (Mountain Floor 2) - Mountain Floor 2 Elevator Room - 0x09EDD:
```

**Review by Berserker66 — APPROVED** [review]

File: worlds/witness/data/WitnessLogicVariety.txt [hunk 6]
```diff
@@ -0,0 +1,1223 @@
...
+158158 - 0x021B7 (Back First Row 3) - 0x021B6 - Shapers & Negative Shapers & Eraser
+158159 - 0x021BB (Back First Row 4) - 0x021B7 - Shapers & Negative Shapers & Eraser
+158160 - 0x09DB5 (Back First Row 5) - 0x021BB - Shapers & Negative Shapers & Eraser
+158161 - 0x09DB1 (Back First Row 6) - 0x09DB5 - Stars & Stars + Same Colored Symbol & Eraser & Colored Squares
+158162 - 0x3C124 (Back First Row 7) - 0x09DB1 - Stars & Stars + Same Colored Symbol & Eraser & Colored Squares
+158163 - 0x09DB3 (Back First Row 8) - 0x3C124 - Stars & Stars + Same Colored Symbol & Eraser & Colored Squares & Triangles
+158164 - 0x09DB4 (Back First Row 9) - 0x09DB3 - Stars & Stars + Same Colored Symbol & Eraser & Colored Squares & Triangles
+158165 - 0x275FA (Hook Control) - True - Shapers & Eraser
+158167 - 0x0A3CB (Back Second Row 1) - 0x09DB4 - Black/White Squares & Colored Squares & Eraser & Shapers
+158168 - 0x0A3CC (Back Second Row 2) - 0x0A3CB - Stars & Stars + Same Colored Symbol & & Eraser & Shapers
```
Comment (Exempt-Medic) [inline worlds/witness/data/WitnessLogicVariety.txt:349]: Typo (it was mine)
```suggestion
158168 - 0x0A3CC (Back Second Row 2) - 0x0A3CB - Stars & Stars + Same Colored Symbol & Eraser & Shapers
```

Comment (NewSoupVi) [reply worlds/witness/data/WitnessLogicVariety.txt:349]: My ass just forgot to `git push` yesterday lmao

**Pull request merged**